src/components/dar_table/DarTable.js

```diff
@@ -87,21 +88,26 @@ export const tableHeaderTemplate = [
   div({style: Styles.TABLE.ELECTION_STATUS_CELL, className: 'cell-sort'}, [
     "Election Status",
     span({ className: 'glyphicon sort-icon glyphicon-sort' })
-  ]),
-  div({style: Styles.TABLE.ELECTION_ACTIONS_CELL}, ["Election Actions"])
+  ])
 ];
+
+export const tableHeaderTemplate = soTableHeaderTemplate.concat(
+  div({style: Styles.TABLE.ELECTION_ACTIONS_CELL}, ["Election Actions"]));
+
```

Comment:
Nit, naming. It looks like we have a set of columns that we're defining as SO columns, then concatenating the election action columns on top of that. By defining the root set of columns as SO columns, we're putting them into a role-based category which isn't how we want to be thinking of them. This will become more convoluted as we look at other role-specific headers, i.e. researchers and data custodians come to mind.

I think this could use some brainstorming. One idea is if `tableHeaderTemplate` were a function, we could pass in an optional role and then granularly build up the divs one by one with checks on the roles.

Reply:
> Nit, naming. It looks like we have a set of columns that we're defining as SO columns, then concatenating the election action columns on top of that. By defining the root set of columns as SO columns, we're putting them into a role-based category which isn't how we want to be thinking of them. This will become more convoluted as we look at other role-specific headers, i.e. researchers and data custodians come to mind.
> 
> I think this could use some brainstorming. One idea is if `tableHeaderTemplate` were a function, we could pass in an optional role and then granularly build up the divs one by one with checks on the roles.

We could make the cells an opt-in sort of thing by roles. Role constants would store an array of key names (that correspond to specific cells) that would be passed into a function. That function would then use the key names to get the appropriate template cells (stored in an const object in the function) and process them to produce the template.

Reply:
Okay I just turned both of the header and row templates into functions that render only the appropriate columns based on consoleType